content/data-structures/SegmentTree.h

```diff
@@ -0,0 +1,38 @@
+/**
+ * Author: Sachin Sivakumar
+ * Date: 2024-10-09
+ * License: CC0
+ * Source: Inspired by KACTL
+ * Description: Segment tree supporting Range Minimum Queries and Point Add Queries
+ * Time: O(\log N)
```

Comment:
Documentation feels incomplete to me. Flesh out a bit more what a segment tree does and I also think maybe time complexity needs an update as well? Doesn't feel descriptive either